components/grouper/src/main/java/org/datacleaner/components/group/GrouperTransformer.java

```diff
@@ -0,0 +1,286 @@
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.components.group;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.ConcurrentMap;
+
+import javax.inject.Named;
+
+import org.apache.metamodel.query.Query;
+import org.apache.metamodel.schema.ColumnType;
+import org.apache.metamodel.util.HasName;
+import org.apache.metamodel.util.ObjectComparator;
+import org.datacleaner.api.Categorized;
+import org.datacleaner.api.Close;
+import org.datacleaner.api.Configured;
+import org.datacleaner.api.Initialize;
+import org.datacleaner.api.InputColumn;
+import org.datacleaner.api.InputRow;
+import org.datacleaner.api.MappedProperty;
+import org.datacleaner.api.MultiStreamComponent;
+import org.datacleaner.api.OutputDataStream;
+import org.datacleaner.api.OutputRowCollector;
+import org.datacleaner.components.categories.CompositionCategory;
+import org.datacleaner.job.output.OutputDataStreamBuilder;
+import org.datacleaner.job.output.OutputDataStreams;
+
+@Named("Grouper")
+@Categorized(value = CompositionCategory.class)
+public class GrouperTransformer extends MultiStreamComponent {
+
+    public static final String PROPERTY_GROUP_KEY = "Group key";
+    public static final String PROPERTY_AGGREGATED_VALUES = "Aggregated values";
+    public static final String PROPERTY_AGGREGATION_TYPES = "AggregationTypes";
+    public static final String PROPERTY_VALUE_SORTATION = "Value sortation";
+
+    private static final Object NULL_KEY = new Object();
+
+    public static enum AggregationType implements HasName {
+        CONCAT_VALUES("Concatenate values"), FIRST_VALUE("Select first value"), CREATE_LIST("Create list of values");
+
+        private final String _name;
+
+        private AggregationType(String name) {
+            _name = name;
+        }
+
+        @Override
+        public String getName() {
+            return _name;
+        }
+    }
+
+    public static enum SortationType implements HasName {
+        NONE("None"), RECORD_ORDER("Record order"), NATURAL_SORT_ASC("Natural sort, ascending"), NATURAL_SORT_DESC(
+                "Natural sort, descending");
+
+        private final String _name;
+
+        private SortationType(String name) {
+            _name = name;
+        }
+
+        @Override
+        public String getName() {
+            return _name;
+        }
+    }
+
+    @Configured(order = 1, value = PROPERTY_GROUP_KEY)
```

Comment:
How will the UI respond to this being first? If I'm not wrong, it would try to link this instead of the aggregated values, which seems wrong.

Or do I remember this wrongly, and the *@*MappedProperty types make it impossible to link using the UI anyway? Then I guess this is the best way to get a bit of the functionality back.